src/main/java/uk/gov/ons/census/fwmt/jobservice/converter/impl/CCSIVConverter.java

```diff
@@ -17,21 +16,31 @@
 import uk.gov.ons.census.fwmt.common.data.modelcase.ModelCase;
 import uk.gov.ons.census.fwmt.common.error.GatewayException;
 import uk.gov.ons.census.fwmt.jobservice.converter.CometConverter;
-import uk.gov.ons.census.fwmt.jobservice.entity.CCSOutcomeEntity;
-import uk.gov.ons.census.fwmt.jobservice.service.JobCacheManager;
-
-import java.io.IOException;
+import uk.gov.ons.census.fwmt.jobservice.entity.CCSOutcomeStore;
+import uk.gov.ons.census.fwmt.jobservice.message.MessageConverter;
 
 import static uk.gov.ons.census.fwmt.jobservice.utils.JobServiceUtils.setAddress;
 
 @Component("CCS")
 public class CCSIVConverter implements CometConverter {
 
   @Autowired
-  private ObjectMapper mapper;
+  private CCSOutcomeStore ccsOutcomeStore;
+
+  @Autowired
+  private MessageConverter messageConverter;
+
+  @Autowired
+  private ObjectMapper objectMapper;
 
   @Override
   public CaseRequest convert(CreateFieldWorkerJobRequest ingest) throws GatewayException {
+//    Address address = new Address();
+//    CeDetails ceDetails = new CeDetails();
+
+    String output = ccsOutcomeStore.retrieveCache(String.valueOf(ingest.getCaseId()));
```

Comment:
Put these 2 lines with the try n catch and then extract it to a private method cached something like getCachedOutcomeDetails() and set the result of this into caseRequest.setSpecialInstructions()

Reply:
I have moved the lines into a try catch that is further down in the code related to setting the special instruction. I have also created a private method for this. 

As discussed, the message needs to be converted into a smaller object, otherwise we end up with a lot of redundant null values.